var/spack/repos/builtin/packages/py-ipykernel/package.py

```diff
@@ -25,7 +25,8 @@ class PyIpykernel(PythonPackage):
     version('4.1.0', '638a43e4f8a15872f749090c3f0827b6')
 
     depends_on('python@2.7:2.8,3.3:')
-    depends_on('py-setuptools', type='build')
+    depends_on('python@3.3:', when='@5:')
+    depends_on('py-setuptools', type='build', when='@5:')
     depends_on('py-traitlets@4.1.0:', type=('build', 'run'))
     depends_on('py-tornado@4.0:', when='@:4.999', type=('build', 'run'))
     depends_on('py-tornado@4.2:', when='@5.0.0:', type=('build', 'run'))
```

Comment:
Is this `py-pexpect` dependency at the bottom actually required? I don't see it listed in the `setup.py`

Reply:
according to the comment in this [PR](https://github.com/ipython/ipykernel/commit/5c8ef17623faa986f9597bc74b0afe3bfd2e9490) they do on posix, but ship their own if no system one is available

Reply:
we have it as a dependency since the original commit 8c650303ffdd02c4be273efbce1674706c0182cd so I'd say just leave it in?